lib/matplotlib/cbook/__init__.py

```diff
@@ -2766,3 +2766,42 @@ def _topmost_artist(
     in reverse order.
     """
     return _cached_max(reversed(artists))
+
+
+def _define_aliases(local_d, alias_d):
+    """Define property aliases.
+
+    Use in a class definition as ::
+
+        cbook._define_aliases(locals(), {
+            "property": ["alias", ...], ...
+        })
+
+    For each property, if the corresponding ``get_property`` is defined in the
+    class so far, an alias named ``get_alias`` will be defined; the same will
+    be done for setters.  If neither the getter nor the setter exists, an
+    exception will be raised.
+
+    The alias map is stored as the ``_alias_map`` attribute on the class and
+    can be used by `~.normalize_kwargs`.
+    """
+
+    def make_alias(name):  # Enfore a closure over *name*.
+        def method(self, *args, **kwargs):
+            return getattr(self, name)(*args, **kwargs)
+        method.__doc__ = "alias for {}".format(name)
+        return method
+
+    for prop, aliases in alias_d.items():
+        exists = False
+        for prefix in ["get_", "set_"]:
+            if prefix + prop in local_d:
+                exists = True
+                for alias in aliases:
+                    method = make_alias(prefix + prop)
+                    method.__name__ = str(prefix + alias)  # Py2 compat.
```

Comment:
I think this line should be handled by `make_alias(orig_name, alias_name)`

Reply:
I'd rather make make_alias not know what the name of the resulting function is.  Really its only goal is to make a closure over the name (so I'm even going to move the `__doc__` out, in fact).

Reply:
I think you should close over the value, not the name

Reply:
Can you explain what the py2 compat issue is? Seems that this doesn't matter for anything other than printing the `__name__`, which py3 does too.

Reply:
py2 does not want unicode strings as function names, which we do end up passing because of the unicode_literals future alias.  There's a similar issue with the first argument to subprocess on Windows + Py2, again we would pass in unicode strings because of unicode_literals, so we have to wrap them in a call to str().